code/entities/cooking/hotdog-roller/interactable/Roller.cs

```diff
@@ -0,0 +1,122 @@
+﻿using Conna.Inventory;
+using Sandbox;
+using Sandbox.util;
+using System.Collections.Generic;
+
+namespace Cinema.HotdogRoller;
+
+public class Roller : BaseInteractable
+{
+    private struct Slot
+    {
+        public HotdogCookable Entity { get; set; }
+        public string Attachment { get; set; }
+
+        public Slot(string attachment)
+        {
+            Attachment = attachment;
+        }
+    }
+
+    private struct SlotDist
+    {
+        public Slot Slot { get; set; }
+        public float Distance { get; set; }
+        public int Index { get; set; }
+        public Transform SlotTransform { get; set; }
+
+        public SlotDist(Slot slot, float dist, int index, Transform t)
+        {
+            Slot = slot;
+            Distance = dist;
+            Index = index;
+            SlotTransform = t;
+        }
+    }
+
+    public Knob Knob { get; set; }
+    public Switch Switch { get; set; }
+    private bool front;
+    private Slot[] slots = new Slot[10];
+
+    public Roller() // For the compiler...
+    {
+    }
+
+    public Roller(Switch p_switch, Knob knob, bool front = false)
```

Comment:
Nit: Please avoid python style naming , `pSwitch`

Also, what is `pSwitch` ? Is `p` an abbreviation and could you please spell it out